### ISSUE TITLE
java.lang.RuntimeException: Methods marked with @UiThread must be executed on the main thread

### ISSUE THREAD

**Issue by bardram** [dooboolab/flutter_sound#93]
### Version of flutter_sound

*1.4.1*

## flutter doctor

````
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel master, v1.6.8-pre.33, on Mac OS X 10.13.6 17G7024, locale en-DK)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[✗] Xcode - develop for iOS and macOS
    ✗ Xcode installation is incomplete; a full installation is necessary for iOS development.
      Download at: https://developer.apple.com/xcode/download/
      Or install Xcode via the App Store.
      Once installed, run:
        sudo xcode-select --switch /Applications/Xcode.app/Contents/Developer
    ✗ CocoaPods not installed.
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To install:
        brew install cocoapods
        pod setup
[✗] iOS tools - develop for iOS devices
    ✗ libimobiledevice and ideviceinstaller are not installed. To install with Brew, run:
        brew update
        brew install --HEAD usbmuxd
        brew link usbmuxd
        brew install --HEAD libimobiledevice
        brew install ideviceinstaller
    ✗ ios-deploy not installed. To install:
        brew install ios-deploy
[✓] Android Studio (version 3.4)
[✓] IntelliJ IDEA Ultimate Edition (version 2017.3.5)
[✓] Connected device (1 available)

! Doctor found issues in 2 categories.
````

### Platforms you faced the error (IOS or Android or both?)

Android

### Expected behavior

Just testing the example app.

### Actual behavior

When starting the recording, the following exception happens:

````
D/FlutterSoundPlugin(10473): startRecorder
I/MediaRecorderJNI(10473): setup
I/MediaRecorderJNI(10473): setAudioSource(1)
I/MediaRecorderJNI(10473): setAudioEncoder(3)
I/MediaRecorderJNI(10473): setParameter()
I/MediaRecorderJNI(10473): setParameter()
I/MediaRecorderJNI(10473): setOutputFile
I/MediaRecorderJNI(10473): prepare
I/MediaRecorderJNI(10473): start
E/FlutterSoundPlugin(10473): Exception: 
E/FlutterSoundPlugin(10473): java.lang.RuntimeException: Methods marked with @UiThread must be executed on the main thread. Current thread: pool-1-thread-1
E/FlutterSoundPlugin(10473): 	at io.flutter.embedding.engine.FlutterJNI.ensureRunningOnMainThread(FlutterJNI.java:605)
E/FlutterSoundPlugin(10473): 	at io.flutter.embedding.engine.FlutterJNI.invokePlatformMessageResponseCallback(FlutterJNI.java:556)
E/FlutterSoundPlugin(10473): 	at io.flutter.embedding.engine.dart.DartMessenger$Reply.reply(DartMessenger.java:131)
E/FlutterSoundPlugin(10473): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler$1.success(MethodChannel.java:225)
E/FlutterSoundPlugin(10473): 	at com.dooboolab.fluttersound.FlutterSoundPlugin.startRecorder(FlutterSoundPlugin.java:198)
E/FlutterSoundPlugin(10473): 	at com.dooboolab.fluttersound.FlutterSoundPlugin.lambda$onMethodCall$0$FlutterSoundPlugin(FlutterSoundPlugin.java:65)
E/FlutterSoundPlugin(10473): 	at com.dooboolab.fluttersound.-$$Lambda$FlutterSoundPlugin$Uw7hmMVnMhHSlTmtp8I_WAUlmZE.run(Unknown Source:8)
E/FlutterSoundPlugin(10473): 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:457)
E/FlutterSoundPlugin(10473): 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
E/FlutterSoundPlugin(10473): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
E/FlutterSoundPlugin(10473): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
E/FlutterSoundPlugin(10473): 	at java.lang.Thread.run(Thread.java:764)
````

When hitting "PLAY", this exception happens:

````
V/MediaPlayerNative(10473): message received msg=300, ext1=0, ext2=0
V/MediaPlayerNative(10473): Received SEC_MM_PLAYER_CONTEXT_AWARE
V/MediaPlayerNative(10473): callback application
V/MediaPlayerNative(10473): back from callback
V/MediaPlayerNative(10473): message received msg=100, ext1=1, ext2=-2147483648
E/MediaPlayerNative(10473): error (1, -2147483648)
V/MediaPlayerNative(10473): signal application thread
V/MediaPlayerNative(10473): prepare complete - status=1
E/FlutterSoundPlugin(10473): startPlayer() exception
E/flutter (10473): [ERROR:flutter/lib/ui/ui_dart_state.cc(148)] Unhandled Exception: Exception: PlatformException(ERR_UNKNOWN, ERR_UNKNOWN, Prepare failed.: status=0x1)
E/flutter (10473): #0      FlutterSound.startPlayer (package:flutter_sound/flutter_sound.dart:165:7)
E/flutter (10473): <asynchronous suspension>
E/flutter (10473): #1      _MyAppState.startPlayer (package:flutter_sound_example/main.dart:98:38)
E/flutter (10473): <asynchronous suspension>
E/flutter (10473): #2      _MyAppState.build.<anonymous closure> (package:flutter_sound_example/main.dart:234:25)
E/flutter (10473): #3      _InkResponseState._handleTap (package:flutter/src/material/ink_well.dart:635:14)
E/flutter (10473): #4      _InkResponseState.build.<anonymous closure> (package:flutter/src/material/ink_well.dart:711:32)
E/flutter (10473): #5      GestureRecognizer.invokeCallback (package:flutter/src/gestures/recognizer.dart:182:24)
E/flutter (10473): #6      TapGestureRecognizer._checkUp (package:flutter/src/gestures/tap.dart:365:11)
E/flutter (10473): #7      TapGestureRecognizer.acceptGesture (package:flutter/src/gestures/tap.dart:312:7)
E/flutter (10473): #8      GestureArenaManager.sweep (package:flutter/src/gestures/arena.dart:156:27)
E/flutter (10473): #9      _WidgetsFlutterBinding&BindingBase&GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:222:20)
E/flutter (10473): #10     _WidgetsFlutterBinding&BindingBase&GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:198:22)
E/flutter (10473): #11     _WidgetsFlutterBinding&BindingBase&GestureBinding._handlePointerEvent (package:flutter/src/gestures/binding.dart:156:7)
E/flutter (10473): #12     _WidgetsFlutterBinding&BindingBase&GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:102:7)
E/flutter (10473): #13     _WidgetsFlutterBinding&BindingBase&GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:86:7)
E/flutter (10473): #14     _rootRunUnary (dart:async/zone.dart:1136:13)
E/flutter (10473): #15     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
E/flutter (10473): #16     _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)
E/flutter (10473): #17     _invoke1 (dart:ui/hooks.dart:239:10)
E/flutter (10473): #18     _dispatchPointerDataPacket (dart:ui/hooks.dart:159:5)
E/flutter (10473): 

````

### Tested environment (Emulator? Real Device?)

Real Device : Android, Samsung.

### Steps to reproduce the behavior

Just run the example app.

**Comment by bardram** [comment]
A similar exception has been reported in many other Flutter plugins lately. See e.g.

* https://github.com/flutter/flutter/issues/32315
* https://github.com/g123k/flutter_plugin_device_apps/issues/11

**Comment by nativeaditya** [comment]
same issue any solution?

**Comment by xiaoshitou123** [comment]
same issue any solution?

**Comment by zhcAndroid** [comment]
同样的问题有什么办法

**Comment by zhcAndroid** [comment]
不能在thread中执行 result接口中方法
使用handler.post切换回主线成即可
code如下：

      String finalPath = path;
      mainHandler.post(new Runnable() {
        @Override
        public void run() {
          result.success(finalPath);
        }
      });

**Comment by hyochan** [comment]
Fixed in #107

**Comment by rowild** [comment]
@zhcAndroid My friend - why not in English? So all can understand??
